### DISCUSSION TITLE
Use of distinctUntilChanged

### DISCUSSION THREAD

**Discussion by raghunandankavi2010** [PatilShreyas/Foodium#61] [General]
Should we not use `distinctUntilChanged()` in dao?.

      @Query("SELECT * FROM ${Post.TABLE_NAME} WHERE ID = :postId")
      fun getPostById(postId: Int): Flow<Post>

      fun getPostByIdUntilChanged(id: Int) = getPostById(id).distinctUntilChanged()

cause if we update a unrelated row the query also emits data with the same result again. https://medium.com/androiddevelopers/room-flow-273acffe5b57

**Comment by PatilShreyas** [comment]
@raghunandankavi2010 I think this makes sense. Need to implement this in the Foodium app.

**Reply by PatilShreyas** [reply]
created separate issue for this: https://github.com/PatilShreyas/Foodium/issues/68
